common/models/report-model.js

```diff
@@ -1615,6 +1615,34 @@ module.exports = function (ReportModel) {
             });
     }
 
+    ReportModel.regenerateOrder = function (orgModelId, reportModelId, options){
+        logger.debug({
+            message: "will change state to generated",
+            orgModelId,
+            reportModelId,
+            options,
+            functionName: 'regenerateOrder'
+        });
+        var reportModelInstance;
+        return ReportModel.findById(reportModelId)
+            .catch(function (error) {
+                logger.error({
+                    message: 'Could not find report model instance',
+                    reportModelId,
+                    options,
+                    error,
+                    functionName: 'regenerateOrder'
+                });
+                return Promise.reject('Could not find report model instance');
+            })
+            .then( function(response) {
+                reportModelInstance = response;
+                return reportModelInstance.updateAttributes({
+                    state: REPORT_STATES.GENERATED
+                });
+            })
```

Comment:
might need a catch block at the end

Reply:
it's there at 1628, followed the same code convention as used earlier in the file.